src/lib.rs

```diff
@@ -431,4 +474,15 @@ impl<S: ?Sized + LanguageServer> LanguageServer for Box<S> {
     fn document_highlight(&self, params: TextDocumentPositionParams) -> Self::HighlightFuture {
         (**self).document_highlight(params)
     }
+
+    fn signature_help(&self, params: TextDocumentPositionParams) -> Self::SignatureHelpFuture {
+        (**self).signature_help(params)
+    }
+
+    fn goto_implementation(
```

Comment:
This method should be moved to immediately below `LanguageServer::goto_type_definition()`.